test/payment_icon_name_shitlist.rb

```diff
@@ -0,0 +1,96 @@
+# Dated: 9th Nov 2022
+# Author: Rahil Virani
+#
+# The special characters in the name has caused a lot of issues in the past
+# to seamlessly deploy the new icons. This is casued by
+# repositories using different seperators for their local list of acceptable icons
+# i.e. Jascript uses camelcase vs. others using underscore. The value of the name
+# also reflects the name of the svg file and certain attributes in the svg, it is becoming
+# harder to keep them in sync in multiple repositories.
+#
+# Decision: Moving forward we should only accept lower case alpha characters in the name
+class PaymentIconNameShitlist
```

Comment:
So this class is going to be used as a validation test only? Not sure if `Shitlist` is the most meaningful name for that purpose 😅

Reply:
If we see this class as a repository of icon names that we will likely do something on the future, i.e fix things and remove icons from it, then I would be fine with `PaymentIconNameShitlist` since `shitlists` are frequently used in Shopify.

But if this is more of a permanent solution, maybe `PaymentIconLegacyNames` make more sense.